utils/tinystr/src/unvalidated.rs

```diff
@@ -5,16 +5,27 @@
 use crate::TinyAsciiStr;
 use crate::TinyStrError;
 
+/// A fixed-length bytes array that is expected to be an ASCII string but does not enforce that invariant.
+///
+/// Use this type instead of `TinyAsciiStr` if you don't need to enforce ASCII during deserialization. For
+/// example, strings that are keys of a map don't need to ever be reified as `TinyAsciiStr`s.
+///
+/// The main advantage of this type over `[u8; N]` is that it serializes as a string in
+/// human-readable formats like JSON.
 #[derive(Debug, PartialEq, PartialOrd, Eq, Ord, Clone, Copy)]
 pub struct UnvalidatedTinyAsciiStr<const N: usize>(pub(crate) [u8; N]);
 
 impl<const N: usize> UnvalidatedTinyAsciiStr<N> {
+    #[inline]
+    // Converts into a [`TinyAsciiStr`]. Fails if the bytes are not valid ASCII.
     pub fn try_into_tinystr(&self) -> Result<TinyAsciiStr<N>, TinyStrError> {
```

Comment:
thought: could also be called `try_into_validated()`

(or just a TryInto impl?)

Reply:
`TryInto` isn't as chainable, I added `as_tinystr` to the subtags in the PR for the same reason.

Reply:
What do you mean by `chainable` here? `TryInto` seems like the right trait for me and it abstracts better for generic inputs.

Reply:
I think he's talking about how inference does not succeed if you do stuff like `foo.try_into().bar()`

Reply:
We definitely should _have_ `TryInto`, IMO it's acceptable to also have an inherent impl.

Reply:
Ah, ok, for any further readers, Robert meant this: `TryInto::<Bar>::try_into(Foo {})?.baz();` being ugly.